src/tracing/builder/geth.rs

```diff
@@ -169,6 +172,18 @@ impl GethTraceBuilder {
         }
     }
 
+    /// Returns true if the given trace or any of its parents failed.
+    fn call_or_parent_failed(&self, node: &CallTraceNode) -> bool {
+        if node.trace.is_error() {
+            return true;
+        }
+
+        if let Some(parent) = node.parent {
+            return self.call_or_parent_failed(&self.nodes[parent]);
+        }
+        false
```

Comment:
can we solve this iteratively by looping instead?

Reply:
done